Update the arp update test failure

nhe-NV opened the pull request:
The ip used in the test case is hard coded with the base ip + 2, in some topo, the ip is not added to the arp responder, then the test could fail. update it to use dynamic offset which could make sure the target ip in in the arp responder.

<!--
Please make sure you've read and understood our contributing guidelines;
https://github.com/sonic-net/SONiC/blob/gh-pages/CONTRIBUTING.md

Please provide following information to help code review process a bit easier:
-->
### Description of PR
<!--
- Please include a summary of the change and which issue is fixed.
- Please also include relevant motivation and context. Where should reviewer start? background context?
- List any dependencies that are required for this change.
-->

Summary:
Fixes # (issue)

### Type of change

<!--
- Fill x for your type of change.
- e.g.
- [x] Bug fix
-->

- [x] Bug fix
- [ ] Testbed and Framework(new/improvement)
- [ ] New Test case
    - [ ] Skipped for non-supported platforms
- [ ] Test case improvement


### Back port request
- [ ] 202012
- [ ] 202205
- [ ] 202305
- [ ] 202311
- [ ] 202405
- [x] 202411

### Approach
#### What is the motivation for this PR?

#### How did you do it?

#### How did you verify/test it?

#### Any platform specific information?

#### Supported testbed topology if it's a new test case?

### Documentation
<!--
(If it's a new feature, new test case)
Did you update documentation/Wiki relevant to your implementation?
Link to the wiki page?
-->

mssonicbld commented:
/azp run

nhe-NV commented:
/azpw run

mssonicbld commented:
/AzurePipelines run

bingwang-ms commented on tests/common/fixtures/ptfhost_utils.py:
```diff
@@ -231,7 +231,7 @@ def setup_vlan_arp_responder(ptfhost, rand_selected_dut, tbinfo):
     ptfhost.command('supervisorctl start arp_responder')
 
-    yield vlan, ipv4_base, ipv6_base
+    yield vlan, ipv4_base, ipv6_base, ip_offset
```
Since you changed the signature of function `setup_vlan_arp_responder`, please do a global search to ensure all the references are updated.

nhe-NV replied:
@bingwang-ms the fixture "setup_vlan_arp_responder" is only used in the tests/arp/test_arp_update.py currently.

mssonicbld commented:
/azp run

nhe-NV commented:
/azpw run

mssonicbld commented:
/AzurePipelines run

mssonicbld commented:
Cherry-pick PR to 202411: https://github.com/sonic-net/sonic-mgmt/pull/17485

lolyu reviewed (commented):
this is causing a regression on dualtor testbed

nhe-NV commented:
> this is causing a regression on dualtor testbed:
> 
> ```
>         for port in vlan_members:
>             ptf_index = dut_to_ptf_port_map[port]
>             if 'dualtor' in tbinfo['topo']['name']:
>                 arp_responder_cfg['eth{}'.format(ptf_index)] = [
>                     server_ip[port]['server_ipv4'].split('/')[0],
>                     server_ip[port]['server_ipv6'].split('/')[0]
>                 ]
>                 continue
>             ip_offset = ptf_index + 1  # Add one since PTF indices start at 0
>             arp_responder_cfg['eth{}'.format(ptf_index)] = [
>                 str(ipv4_base.ip + ip_offset), str(ipv6_base.ip + ip_offset)
>             ]
>     
>         CFG_FILE = '/tmp/arp_responder_vlan.json'
>         with open(CFG_FILE, 'w') as file:
>             json.dump(arp_responder_cfg, file)
>         ptfhost.copy(src=CFG_FILE, dest=CFG_FILE)
>     
>         extra_vars = {
>                 'arp_responder_args': '--conf {}'.format(CFG_FILE)
>         }
>     
>         ptfhost.host.options['variable_manager'].extra_vars.update(extra_vars)
>         ptfhost.template(src='templates/arp_responder.conf.j2',
>                          dest='/etc/supervisor/conf.d/arp_responder.conf')
>         ptfhost.copy(src=os.path.join(SCRIPTS_SRC_DIR, ARP_RESPONDER_PY),
>                      dest=OPT_DIR)
>     
>         ptfhost.command('supervisorctl reread')
>         ptfhost.command('supervisorctl update')
>     
>         logger.info("Start arp_responder")
>         ptfhost.command('supervisorctl start arp_responder')
>     
> >       yield vlan, ipv4_base, ipv6_base, ip_offset
> E       UnboundLocalError: local variable 'ip_offset' referenced before assignment
> ```

This is due to not tested the code together with PR https://github.com/sonic-net/sonic-mgmt/pull/16948, since the 2 PRs opened almost at same day, I will handle and open a new PR

nhe-NV commented:
@lolyu Here is the PR for the fix, can you help to merge? https://github.com/sonic-net/sonic-mgmt/pull/17543
